ydb/core/tx/datashard/datashard_ut_write.cpp

```diff
@@ -716,5 +719,452 @@ Y_UNIT_TEST_SUITE(DataShardWrite) {
 
     }  // Y_UNIT_TEST
 
+    using TWriteRequestPtr = std::unique_ptr<NEvents::TDataEvents::TEvWrite>;
+    using TModifyWriteRequestCallback = std::function<void(const TWriteRequestPtr&)>;
+
+    void PrepareMultiShardWrite(
+            TTestActorRuntime& runtime,
+            const TActorId& sender,
+            const TTableId& tableId,
+            ui64 txId, ui64& coordinator, ui64& minStep, ui64& maxStep,
+            ui64 shardId,
+            i32 key,
+            i32 value,
+            ui64 arbiterShard,
+            const std::vector<ui64>& sendingShards,
+            const std::vector<ui64>& receivingShards,
+            const TModifyWriteRequestCallback& modifyWriteRequest = [](auto&){})
+    {
+        Cerr << "========= Sending prepare to " << shardId << " =========" << Endl;
+
+        auto req = std::make_unique<NEvents::TDataEvents::TEvWrite>(txId, NKikimrDataEvents::TEvWrite::MODE_VOLATILE_PREPARE);
+
+        const std::vector<ui32> columnIds({ 1, 2 });
+
+        TVector<TCell> cells;
+        cells.push_back(TCell::Make(key));
+        cells.push_back(TCell::Make(value));
+        TSerializedCellMatrix matrix(cells, 1, columnIds.size());
+        TString data = matrix.ReleaseBuffer();
+        const ui64 payloadIndex = req->AddPayload(TRope(std::move(data)));
+        req->AddOperation(NKikimrDataEvents::TEvWrite_TOperation::OPERATION_UPSERT, tableId, columnIds, payloadIndex, NKikimrDataEvents::FORMAT_CELLVEC);
+
+        auto* kqpLocks = req->Record.MutableLocks();
+        kqpLocks->SetOp(NKikimrDataEvents::TKqpLocks::Commit);
+        if (arbiterShard) {
+            kqpLocks->SetArbiterShard(arbiterShard);
+        }
+        for (ui64 sendingShard : sendingShards) {
+            kqpLocks->AddSendingShards(sendingShard);
+        }
+        for (ui64 receivingShard : receivingShards) {
+            kqpLocks->AddReceivingShards(receivingShard);
+        }
+
+        modifyWriteRequest(req);
+
+        SendViaPipeCache(runtime, shardId, sender, std::move(req));
+
+        auto ev = runtime.GrabEdgeEventRethrow<NEvents::TDataEvents::TEvWriteResult>(sender);
+        auto* msg = ev->Get();
+        UNIT_ASSERT_VALUES_EQUAL(msg->Record.GetStatus(), NKikimrDataEvents::TEvWriteResult::STATUS_PREPARED);
+        minStep = Max(minStep, msg->Record.GetMinStep());
+        maxStep = Min(maxStep, msg->Record.GetMaxStep());
+        UNIT_ASSERT_VALUES_EQUAL(msg->Record.DomainCoordinatorsSize(), 1);
+        if (coordinator == 0) {
+            coordinator = msg->Record.GetDomainCoordinators(0);
+        } else {
+            UNIT_ASSERT_VALUES_EQUAL(coordinator, msg->Record.GetDomainCoordinators(0));
+        }
+    }
+
+    void SendProposeToCoordinator(
```

Comment:
There is a similar method [SendProposeToCoordinator](https://github.com/ydb-platform/ydb/blob/293e40077ccd74eba4b96a089fb830a918dea1b3/ydb/core/tx/datashard/ut_common/datashard_ut_common.cpp#L2101)

It should be better either use a single method, or move this overloaded method to datashard_ut_common.cpp

Reply:
I prefer local functions unless they are useful in multiple translation units. The existing function may have the same name, but it has the wrong arguments and doesn't work correctly, fixing existing tests was out of scope of this pr.